.github/workflows/clean_cache.yml

```diff
@@ -0,0 +1,33 @@
+on:
+  pull_request:
+    branches: 
+      - master
+    types:
+      - closed
+
+jobs:
+  clear_cache:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install and setup mc client
+        run: |
+          wget https://dl.min.io/client/mc/release/linux-amd64/mc
+          chmod +x mc
+          mc --help
+          mc alias set hetzner https://hel1.your-objectstorage.com {{ secrets.HETZNER_S3_ACCESS_KEY }} ${{ secrets.HETZNER_S3_ACCESS_SECRET }}
+      
+      - name: Remove matching cache objects
+        run: |
+          # filter objects to ones matching this branch
+          all_objects=$(mc ls --quiet hetzner/gha-cache | awk '{ print $6 }')
+          filtered=()
+          # -n $line: if there's no \newline at the end, read will miss the last line
+          echo $all_objects | while read -r line || [ -n "$line" ]; do
+            if [[ "$line" =~ "^.*{{ github.head_ref }}[-[0-9]*]?$" ]]; then
```

Comment:
for a branch called `nn-blabla` this will match 
- `nn-blabla`
- `nn-blabla-`
- `nn-blabla-1238239`
- `fajl-nn-blabla-1238239`
- or similar

but _not_:
- `nn-blabla-af`
- or similar

i.e. it'd a good idea to prefix branch names with initials or other conventions.

Reply:
are you going to pr this again? if so can you also add `_` to the match, i prefer that as my separator (snake case dies hard)